src/main.rs

```diff
@@ -189,6 +189,14 @@ fn parse_args(config: config::Config) {
         testing: false,
     };
 
+    let mut device_power = DevicePower {
+        did_init: false,
+        _has_battery: false,
+        _best_lid_path: String::new(),
+        _best_power_source_path: String::new(),
+        _best_battery_charge_path: String::new(),
+    };
```

Comment:
DevicePower struct instantiation is duplicated here. I recommend using the Default trait.